build: cross-compiling  for `x86_64-apple-darwin` from `arm64-apple-darwin` broken

Was broken in https://github.com/bitcoin/bitcoin/pull/30997#issuecomment-2772450098:
```bash
make -C depends -j13 HOST=x86_64-apple-darwin
gmake: Entering directory '/Users/bitcoin/depends'
Extracting native_qt...
/Users/bitcoin/depends/sources/qtbase-everywhere-src-6.7.3.tar.xz: OK
/Users/bitcoin/depends/sources/qttranslations-everywhere-src-6.7.3.tar.xz: OK
/Users/bitcoin/depends/sources/qttools-everywhere-src-6.7.3.tar.xz: OK
/Users/bitcoin/depends/sources/CMakeLists.txt-6.7.3: OK
/Users/bitcoin/depends/sources/ECMOptionalAddSubdirectory.cmake-6.7.3: OK
/Users/bitcoin/depends/sources/QtTopLevelHelpers.cmake-6.7.3: OK
Preprocessing native_qt...
patching file 'qtbase/configure'
patching file 'qtbase/bin/qt-cmake.in'
patching file 'qtbase/bin/qt-configure-module.in'
patching file 'qtbase/src/tools/moc/main.cpp'
patching file 'qtbase/cmake/QtBaseGlobalTargets.cmake'
patching file 'qtbase/cmake/QtBaseGlobalTargets.cmake'
patching file 'qtbase/src/CMakeLists.txt'
patching file 'qtbase/src/tools/CMakeLists.txt'
patching file 'qtbase/src/tools/rcc/rcc.cpp'
patching file 'qttools/configure.cmake'
patching file 'qttools/src/linguist/CMakeLists.txt'
Configuring native_qt...
opt: pcre val: qt
opt: zlib val: qt
'/opt/homebrew/bin/cmake' '-DCMAKE_EXE_LINKER_FLAGS=' '-DQT_INTERNAL_CALLED_FROM_CONFIGURE:BOOL=TRUE' '-DBUILD_SHARED_LIBS=OFF' '-DCMAKE_INSTALL_PREFIX=/Users/bitcoin/depends/x86_64-apple-darwin' '-DCMAKE_BUILD_TYPE=Release' '-DINPUT_pkg_config=no' '-DINPUT_reduce_relocations=no' '-DINPUT_use_gold_linker_alias=no' '-DINPUT_concurrent=no' '-DINPUT_network=no' '-DINPUT_printsupport=no' '-DINPUT_sql=no' '-DINPUT_testlib=no' '-DINPUT_xml=no' '-DINPUT_gui=no' '-DINPUT_widgets=no' '-DINPUT_glib=no' '-DINPUT_icu=no' '-DINPUT_libudev=no' '-DINPUT_openssl=no' '-DINPUT_zstd=no' '-DINPUT_pcre=qt' '-DINPUT_system_zlib=no' '-DINPUT_backtrace=no' '-DINPUT_permissions=no' '-DINPUT_process=no' '-DINPUT_settings=no' '-DINPUT_androiddeployqt=no' '-DINPUT_macdeployqt=no' '-DINPUT_windeployqt=no' '-DINPUT_qmake=no' '-DINPUT_linguist=yes' '-DINPUT_assistant=no' '-DINPUT_clang=no' '-DINPUT_clangcpp=no' '-DINPUT_designer=no' '-DINPUT_pixeltool=no' '-DINPUT_qdoc=no' '-DINPUT_qtattributionsscanner=no' '-DINPUT_qtdiag=no' '-DINPUT_qtplugininfo=no' '-G' 'Ninja' '/Users/bitcoin/depends/work/build/x86_64-apple-darwin/native_qt/6.7.3-17d11178efa'
-- The CXX compiler identification is AppleClang 17.0.0.17000013
-- The C compiler identification is AppleClang 17.0.0.17000013
-- The ASM compiler identification is AppleClang
-- Found assembler: /Library/Developer/CommandLineTools/usr/bin/clang
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Checking dependencies of submodule 'qtbase'
-- Checking dependencies of submodule 'qtactiveqt'
-- Checking dependencies of submodule 'qtdeclarative'
-- Checking dependencies of submodule 'qttools'
-- Skipping optional dependency 'qtactiveqt' of 'qttools', because qtactiveqt's CMakeLists.txt couldn't be found.
-- Skipping optional dependency 'qtdeclarative' of 'qttools', because qtdeclarative's CMakeLists.txt couldn't be found.
-- Checking dependencies of submodule 'qttranslations'
-- Configuring submodule 'qtbase'
-- [QtBase] CMAKE_BUILD_TYPE was already explicitly set to: 'Release'
CMake Warning at qtbase/cmake/QtPublicAppleHelpers.cmake:920 (message):
  Qt requires at least version 14 of Xcode, you're building against version .
  Please upgrade.  You can turn this warning into an error by configuring
  with -DQT_FORCE_FATAL_APPLE_SDK_AND_XCODE_CHECK=ON.
Call Stack (most recent call first):
  qtbase/cmake/QtBuildHelpers.cmake:443 (_qt_internal_check_apple_sdk_and_xcode_versions)
  qtbase/cmake/QtBuild.cmake:4 (qt_internal_setup_build_and_global_variables)
  qtbase/cmake/QtSetup.cmake:6 (include)
  qtbase/cmake/QtBuildRepoHelpers.cmake:21 (include)
  qtbase/cmake/QtBuildRepoHelpers.cmake:179 (qt_build_internals_set_up_private_api)
  qtbase/cmake/QtBaseHelpers.cmake:154 (qt_build_repo_begin)
  qtbase/CMakeLists.txt:32 (qt_internal_qtbase_build_repo)


CMake Warning at qtbase/cmake/QtPublicAppleHelpers.cmake:934 (message):
  Qt has only been tested with version 14 of the platform SDK, you're using
  15.4.  This is an unsupported configuration.  You may experience build
  issues, and by using the 15.4 SDK you are opting in to new features that Qt
  has not been prepared for.  Please downgrade the SDK you use to build your
  app to version 14, or configure with -DQT_NO_APPLE_SDK_MAX_VERSION_CHECK=ON
  to silence this warning.
Call Stack (most recent call first):
  qtbase/cmake/QtBuildHelpers.cmake:443 (_qt_internal_check_apple_sdk_and_xcode_versions)
  qtbase/cmake/QtBuild.cmake:4 (qt_internal_setup_build_and_global_variables)
  qtbase/cmake/QtSetup.cmake:6 (include)
  qtbase/cmake/QtBuildRepoHelpers.cmake:21 (include)
  qtbase/cmake/QtBuildRepoHelpers.cmake:179 (qt_build_internals_set_up_private_api)
  qtbase/cmake/QtBaseHelpers.cmake:154 (qt_build_repo_begin)
  qtbase/CMakeLists.txt:32 (qt_internal_qtbase_build_repo)


CMake Error at /opt/homebrew/share/cmake/Modules/CMakeTestOBJCCompiler.cmake:64 (message):
  The Objective-C compiler

    "/Library/Developer/CommandLineTools/usr/bin/clang"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: '/Users/bitcoin/depends/work/build/x86_64-apple-darwin/native_qt/6.7.3-17d11178efa/CMakeFiles/CMakeScratch/TryCompile-e5ZtEf'
    
    Run Build Command(s): /opt/homebrew/bin/ninja -v cmTC_2d8d0
    [1/2] /Library/Developer/CommandLineTools/usr/bin/clang   -x objective-c -arch arm64 -mmacosx-version-min=11.0 -fPIE -MD -MT CMakeFiles/cmTC_2d8d0.dir/testObjCCompiler.m.o -MF CMakeFiles/cmTC_2d8d0.dir/testObjCCompiler.m.o.d -o CMakeFiles/cmTC_2d8d0.dir/testObjCCompiler.m.o -c /Users/bitcoin/depends/work/build/x86_64-apple-darwin/native_qt/6.7.3-17d11178efa/CMakeFiles/CMakeScratch/TryCompile-e5ZtEf/testObjCCompiler.m
    [2/2] : && /Library/Developer/CommandLineTools/usr/bin/clang -arch arm64 -mmacosx-version-min=11.0 -Wl,-search_paths_first -Wl,-headerpad_max_install_names  CMakeFiles/cmTC_2d8d0.dir/testObjCCompiler.m.o -o cmTC_2d8d0   && :
    FAILED: cmTC_2d8d0 
    : && /Library/Developer/CommandLineTools/usr/bin/clang -arch arm64 -mmacosx-version-min=11.0 -Wl,-search_paths_first -Wl,-headerpad_max_install_names  CMakeFiles/cmTC_2d8d0.dir/testObjCCompiler.m.o -o cmTC_2d8d0   && :
    ld: library 'System' not found
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    ninja: build stopped: subcommand failed.
    
  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  qtbase/cmake/QtBuildRepoHelpers.cmake:95 (enable_language)
  qtbase/cmake/QtBuildRepoHelpers.cmake:197 (qt_enable_cmake_languages)
  qtbase/cmake/QtBaseHelpers.cmake:154 (qt_build_repo_begin)
  qtbase/CMakeLists.txt:32 (qt_internal_qtbase_build_repo)

-- Configuring incomplete, errors occurred!
CMake Error at qtbase/cmake/QtProcessConfigureArgs.cmake:1077 (message):
  CMake exited with code 1.
```

--- a/depends/packages/native_qt.mk
+++ b/depends/packages/native_qt.mk
@@ -88,6 +88,10 @@ $(package)_config_opts += -no-feature-qtplugininfo
 
 $(package)_config_env := CC="$$(build_CC)"
 $(package)_config_env += CXX="$$(build_CXX)"
+ifeq ($(build_os),darwin)
+$(package)_config_env += OBJC="$$(build_CC)"
+$(package)_config_env += OBJCXX="$$(build_CXX)"
+endif
 
 $(package)_cmake_opts := -DCMAKE_EXE_LINKER_FLAGS="$$(build_LDFLAGS)"
 ifneq ($(V),)

--- a/depends/packages/native_qt.mk
+++ b/depends/packages/native_qt.mk
@@ -152,5 +152,5 @@ endef
 
 define $(package)_postprocess_cmds
   rm -rf doc/ && \
-  mv -t .. translations/
+  mv translations/ ..
 endef

